eclass/qt5-build.eclass

```diff
@@ -657,7 +658,11 @@ qt5_base_configure() {
 
 	# a forwarding header is no longer created since 5.8, causing the system
 	# config to always be used. bug 599636
-	cp src/corelib/global/qconfig.h include/QtCore/ || die
+	local basedir="${S}/"
+	if ver_test ${PV} -lt 5.14.2 || [[ ${QT5_BUILD_TYPE} == live ]]; then
```

Comment:
live versions don't need `$S`? or are you just trying to preserve the current behavior?

Reply:
that file path does not exist in build dir if that is not $S, and upstream has some `if .git/ exists` style conditional in build system unfortunately...

Reply:
I'm not sure I follow. Are you saying `include/QtCore` does not exist in `QT5_BUILD_DIR` in the *release* case? Maybe we should create the directory hierarchy then? Maybe this workaround is obsolete?

Reply:
`${S}/include` directory does not exist for live ebuilds (see [upstream repository](https://code.qt.io/cgit/qt/qtbase.git/tree/?h=5.15)), but exists for release ebuilds.

Reply:
I tested these changes without this bottom part, and still got

```
src/qdoc/clangcodeparser.cpp:58:10: fatal error: clang-c/Index.h: No such file or directory
   58 | #include <clang-c/Index.h>
      |          ^~~~~~~~~~~~~~~~~
```

Reply:
(That is from qdoc-5.15.0_beta3 and 5.15.9999.)

Reply:
Aside from that, at least the 9999 versions (of Qt packages I normally have installed) seem to work fine still.

Reply:
Qt 5.14.2 also built fine without this `cp` line. Only keeping it for <5.14.2 then.

Reply:
I made a little ooopsies.  I didn't have clang installed (was using 'ebuild' to test 'qdoc' in particular)... but shhh, don't tell anyone.  :V

No wonder I couldn't make 'qdoc' work, no matter what I did.  Regardless, it still doesn't work as is.

Either setting ```QT5_BUILD_DIR="${S}"```, or the following is required (or something completely different):

``` bash
src_configure() {
	mkdir -p "${QT5_BUILD_DIR}"/src/qdoc || die
	qt5_qmake "${QT5_BUILD_DIR}"
	cp "${S}"/src/qdoc/qtqdoc-config.pri "${QT5_BUILD_DIR}"/src/qdoc || die
	qt5-build_src_configure
}
```

Reply:
Also, due to upstream commit 50ef2ac2 [1], a similar workaround is required for most other qttools as well, but it can differ a bit from module to module (or at least between 'qdoc' and the others), so I guess we can't do it in the eclass.  Not that it feels like a real fix still...

1. https://code.qt.io/cgit/qt/qttools.git/commit/?h=5.15&id=50ef2ac2

Reply:
I confirm this works now for qdoc 5.14.2 and 5.14.9999, thanks. Maybe you can take care of 5.15+ ebuilds when this was ACKed @Chiitoo

Reply:
Although I hope we don't run into issues like https://bugs.gentoo.org/498894 again, in case that was related.

Reply:
yeah that's the problem, I don't think you can simply say this isn't needed based on a successful build, unless you can replicate exactly the conditions that triggered the bug. IIRC this was a case of the build system using the (old) installed copy of some header instead of the one being built.

Reply:
should we just keep the previous workaround then, even if it touches `$S`?

Reply:
I'd like to merge this today so it is clear to @Chiitoo in which way to fix the broken >=5.15 qttools subpackages

It is still good to go back to out-of-source build even if it is not perfect with that workaround.